### DISCUSSION TITLE
Transient `Parser Error: Could not find type "PhantomCameraHost" in the current scope.` when switching scenes

### DISCUSSION THREAD

**Discussion by nanodeath** [ramokz/phantom-camera#152] [Question / Help Wanted]
When loading directly from the Godot editor into my main scene (which contains the PhantomCamera), everything works fine.

But when going through the main menu scene -> loading screen scene -> main scene route, around 25% of the time I get

```
Parser Error: Could not find type "PhantomCameraHost" in the current scope.
```

which seems to crash the game.

My loading screen is calling `ResourceLoader.LoadThreadedRequest(levelToLoad, cacheMode: ResourceLoader.CacheMode.Replace);` and `GetTree().ChangeSceneToFile(levelToLoad);` to load the main scene.

Is there something I'm doing wrong?

edit: also it references this line: https://github.com/ramokz/phantom-camera/blob/d2e03d9240e2b8470a388484f32fd372f887bc1a/addons/phantom_camera/scripts/viewfinder/viewfinder.gd#L428

Godot 4.2.1 mono/PhantomCamera 0.5.0.5

**Comment by ramokz** [comment]
As far as I understand it, all custom node types should be initialized, or at least recognized, before the scene even loads without you having to do anything extra. So it's quite peculiar that it says it cannot find type PhantomCameraHost.

Based on your edit addition, it might be that the `viewfinder.gd` is trying to initialize when it shouldn't.
Does the random crash occur from the editor when you play the `Main Menu Scene` and then load the `Main Scene`, or is it happening from a build export?

**Reply by nanodeath** [reply]
> when you play the Main Menu Scene and then load the Main Scene,

This one. I was differentiating between "Play Current Scene" on the Main Menu Scene vs the Main Scene (main game). Haven't tested this with build exports.

**Reply by ramokz** [reply]
Would you per chance be able to create an issue and submit a small sample project, for testing purposes, by copying those scenes that only does those 3 steps, i.e., `main menu scene` → `loading screen` → `main scene`?
Can remove all the assets and scripts that don't relate to it.

It's mainly to have the same setup and logic, as I might otherwise be setting up a scene that differs from yours.
No worries if not!

**Reply by nanodeath** [reply]
I haven't had a chance to build a sample project yet, but I've also observed that I can still trigger this sometimes with jumping directly into the main scene from the editor, though...maybe happens a bit less often.

**Reply by ramokz** [reply]
Based on that, it sounds like the addon script is checking for something that might not have been instantiated yet.
Though it's still very odd that it's saying it can't find the _type_ PhantomCameraHost, rather than the node itself.

I'll try and run some tests on my own as well.

**Reply by ramokz** [reply]
Ran a few more tests, but still can't seem to trigger the error, unfortunately…

If you ever get a chance to make and share a minimal reproduction project, then I should be able to test and find the issue fairly quickly.
